### ISSUE TITLE
Watchtower deletes my Minecraft and Booksonic Dockers from my Synology. It seems to only delete them if there is an update to install.

### ISSUE THREAD

**Issue by Bustout** [containrrr/watchtower#677]
**Describe the bug**
Watchtower deletes my Minecraft and Booksonic Dockers from my Synology. It seems to only delete them if there is an update to install.

**To Reproduce**
Run Wathctower docker on a Synology NAS while other docker containers are running and have an update available to install.

**Expected behavior**
Watchtower should update the existing docker containers that are running and not delete them.

**Screenshots**

**Environment**
 - Synology DS918+ DSM 6.2.3-25426 Update 2
 - Intel Celeron J3455
 - Docker version 18.09.0-0513

**Logs from email**
Task Scheduler has completed a scheduled task.

Task: Watchtower
Start time: Sun, 25 Oct 2020 00:00:02 GMT Stop time: Sun, 25 Oct 2020 00:00:13 GMT Current status: 2 (Interrupted) Standard output/error:
time="2020-10-25T07:00:07Z" level=info msg="Running a one time update."
time="2020-10-25T07:00:08Z" level=warning msg="Failed to retrieve container image info: Error: No such image:
sha256:333738857dc9ab1d7feb2fcff5af0c1893472022d33e884a6377250b3f94f6c7"
time="2020-10-25T07:00:12Z" level=info msg="Found new containrrr/watchtower:latest image (sha256:aed26b939d83a57d3c57685f9a675b53b83910bc49a591b5ad9d005a2d9baa98)"
panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x922a74]

goroutine 1 [running]:
github.com/containrrr/watchtower/pkg/container.Container.runtimeConfig(0x100,
0xc000408de0, 0xc000135ce0, 0x0)
	/home/circleci/repo/pkg/container/container.go:198 +0x74 github.com/containrrr/watchtower/pkg/container.dockerClient.StartContainer(0xc82d00,
0xc00034a680, 0x10001, 0x5a00000100, 0xc000408de0, 0xc000135ce0, 0xc0004263c0, 0x20, 0x0, 0x0)
	/home/circleci/repo/pkg/container/client.go:167 +0x86 github.com/containrrr/watchtower/internal/actions.restartStaleContainer(0xc092260100,
0xc000408de0, 0xc000135ce0, 0xc7c8a0, 0xc000352c00, 0xc000347820, 0x1, 0x2540be400, 0x0)
	/home/circleci/repo/internal/actions/update.go:149 +0xdd github.com/containrrr/watchtower/internal/actions.restartContainersInSortedOrder(0xc000436540,
0x3, 0x4, 0xc7c8a0, 0xc000352c00, 0xc000347820, 0x1, 0x2540be400, 0x0)
	/home/circleci/repo/internal/actions/update.go:119 +0x166 github.com/containrrr/watchtower/internal/actions.Update(0xc7c8a0,
0xc000352c00, 0xc000347820, 0x1, 0x2540be400, 0x0, 0xc00025fce0,
0x7576a5)
	/home/circleci/repo/internal/actions/update.go:61 +0x5eb
github.com/containrrr/watchtower/cmd.runUpdatesWithNotifications(0xc000347820)
	/home/circleci/repo/cmd/root.go:229 +0xbe github.com/containrrr/watchtower/cmd.Run(0x10235c0, 0xc000398000, 0x0, 0x3)
	/home/circleci/repo/cmd/root.go:146 +0x15a github.com/spf13/cobra.(*Command).execute(0x10235c0, 0xc00001e090, 0x3, 0x3, 0x10235c0, 0xc00001e090)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2c2 github.com/spf13/cobra.(*Command).ExecuteC(0x10235c0, 0x406305, 0xc00006a058, 0x0)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2fe
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
github.com/containrrr/watchtower/cmd.Execute()
	/home/circleci/repo/cmd/root.go:57 +0x31
main.main()
	/home/circleci/repo/main.go:13 +0x25

**Additional context**
The process deleted my containers twice. The first time without my knowledge and the second time while I watched to see if it was indeed the watchtower container that was deleting them. When I rebuilt my booksonic container I did so with an updated container that required a new image and when I ran watchtower it did not delete the existing booksonic container, only the Minecraft ones. I have attached all of the logs for watchtower.

[watchtower-logs.zip](https://github.com/containrrr/watchtower/files/5522378/watchtower-logs.zip)

**Comment by piksel** [comment]
This seems to be caused by the fact that the docker API client doesn't return `nil` when it fails to retrieve the images. This means that the warning at https://github.com/containrrr/watchtower/blob/e735cc8111dbc06478a161389c245228d8b70cbb/pkg/container/client.go#L124 is correctly printed, but the image still isn't skipped.

**Comment by piksel** [comment]
I have created a PR that should fix the issue of broken updates. We still don't know WHY the Synology docker cannot retrieve the needed information though, so I guess watchtower might still not be able to update your containers.

**Comment by ElTimablo** [comment]
I'm running into this issue again with random containers. So far, it's deleted the acme companion container for my reverse proxy and my torrent client. I also came back to find that it had shut down almost every single one of my containers, including itself. The remaining containers were so completely mangled that I had to rebuild them completely, losing a significant chunk of data in the process. Issues like this make it nearly impossible to trust Watchtower, and I now hold extreme animosity towards the developers.
